solidity/random-beacon/test/RandomBeaconGovernance.test.ts

```diff
@@ -1963,4 +1968,119 @@ describe("RandomBeaconGovernance", () => {
       }
     )
   })
+
+  describe("beginMinimumAuthorizationUpdate", () => {
+    context("when the caller is not the owner", () => {
+      it("should revert", async () => {
+        await expect(
+          randomBeaconGovernance
+            .connect(thirdParty)
+            .beginMinimumAuthorizationUpdate(123)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when the caller is the owner", () => {
+      let tx
+
+      beforeEach(async () => {
+        tx = await randomBeaconGovernance
+          .connect(governance)
+          .beginMinimumAuthorizationUpdate(123)
+      })
+
+      it("should not update the minimum authorization amount", async () => {
+        expect(await randomBeacon.minimumAuthorization()).to.be.equal(
+          initialMinimumAuthorization
+        )
+      })
+
+      it("should start the governance delay timer", async () => {
+        expect(
+          await randomBeaconGovernance.getRemainingMimimumAuthorizationUpdateTime()
+        ).to.be.equal(24 * 14 * 60 * 60) // 2 weeks
+      })
+
+      it("should emit the MinimumAuthorizationUpdateStarted event", async () => {
+        const blockTimestamp = (await ethers.provider.getBlock(tx.blockNumber))
+          .timestamp
+        await expect(tx)
+          .to.emit(randomBeaconGovernance, "MinimumAuthorizationUpdateStarted")
+          .withArgs(123, blockTimestamp)
+      })
+    })
+  })
+
+  describe("finalizeMinimumAuthorizationUpdate", () => {
+    context("when the caller is not the owner", () => {
+      it("should revert", async () => {
+        await expect(
+          randomBeaconGovernance
+            .connect(thirdParty)
+            .finalizeMinimumAuthorizationUpdate()
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when the update process is not initialized", () => {
+      it("should revert", async () => {
+        await expect(
+          randomBeaconGovernance
+            .connect(governance)
+            .finalizeMinimumAuthorizationUpdate()
+        ).to.be.revertedWith("Change not initiated")
+      })
+    })
+
+    context("when the governance delay has not passed", () => {
+      it("should revert", async () => {
+        await randomBeaconGovernance
+          .connect(governance)
+          .beginMinimumAuthorizationUpdate(123)
+
+        await helpers.time.increaseTime(24 * 13 * 60 * 60) // 13 days
```

Comment:
Nitpicking but I'd do `13 * 24 * 60 * 60` for consistency with the rest of 13 days assertions and given that 13 at the beginning is IMO easier to follow.

Reply:
8c14b8b